lib/matplotlib/backends/backend_gtk4.py

```diff
@@ -0,0 +1,683 @@
+            if state & key_mask:
+                if not (prefix == 'shift' and unikey.isprintable()):
+                    key = f'{prefix}+{key}'
+        return key
+
+    def _draw_rubberband(self, rect):
+        self._rubberband_rect = rect
+        # TODO: Only update the rubberband area.
+        self.queue_draw()
+
+    def draw_func(self, drawing_area, ctx, width, height):
+        self.on_draw_event(self, ctx)
+        self._post_draw(self, ctx)
+
+    def _post_draw(self, widget, ctx):
+        if self._rubberband_rect is None:
+            return
+
+        x0, y0, w, h = self._rubberband_rect
+        x1 = x0 + w
+        y1 = y0 + h
+
+        # Draw the lines from x0, y0 towards x1, y1 so that the
+        # dashes don't "jump" when moving the zoom box.
+        ctx.move_to(x0, y0)
+        ctx.line_to(x0, y1)
+        ctx.move_to(x0, y0)
+        ctx.line_to(x1, y0)
+        ctx.move_to(x0, y1)
+        ctx.line_to(x1, y1)
+        ctx.move_to(x1, y0)
+        ctx.line_to(x1, y1)
+
+        ctx.set_antialias(1)
+        ctx.set_line_width(1)
+        ctx.set_dash((3, 3), 0)
+        ctx.set_source_rgb(0, 0, 0)
+        ctx.stroke_preserve()
+
+        ctx.set_dash((3, 3), 3)
+        ctx.set_source_rgb(1, 1, 1)
+        ctx.stroke()
+
+    def on_draw_event(self, widget, ctx):
+        # to be overwritten by GTK4Agg or GTK4Cairo
+        pass
+
+    def draw(self):
+        # docstring inherited
+        if self.is_drawable():
+            self.queue_draw()
+
+    def draw_idle(self):
+        # docstring inherited
+        if self._idle_draw_id != 0:
+            return
+        def idle_draw(*args):
+            try:
+                self.draw()
+            finally:
+                self._idle_draw_id = 0
+            return False
+        self._idle_draw_id = GLib.idle_add(idle_draw)
+
+    def flush_events(self):
+        # docstring inherited
+        context = GLib.MainContext.default()
+        while context.pending():
+            context.iteration(True)
+
+
+class FigureManagerGTK4(FigureManagerBase):
+    """
+    Attributes
+    ----------
+    canvas : `FigureCanvas`
+        The FigureCanvas instance
+    num : int or str
+        The Figure number
+    toolbar : Gtk.Box
+        The toolbar
+    vbox : Gtk.VBox
+        The Gtk.VBox containing the canvas and toolbar
+    window : Gtk.Window
+        The Gtk.Window
+
+    """
+    def __init__(self, canvas, num):
+        app = _create_application()
+        self.window = Gtk.Window()
+        app.add_window(self.window)
+        super().__init__(canvas, num)
+
+        try:
+            self.window.set_icon_from_file(window_icon)
+        except Exception:
+            # Some versions of gtk throw a glib.GError but not all, so I am not
+            # sure how to catch it.  I am unhappy doing a blanket catch here,
+            # but am not sure what a better way is - JDH
+            _log.info('Could not load matplotlib icon: %s', sys.exc_info()[1])
```

Comment:
This log is emitted, because `'Window' object has no attribute 'set_icon_from_file'` (perhaps we should indeed get rid of the blanket catch, as suggested above, to avoid hiding this).

Reply:
Right, we need to do something completely different on GTK4, for which I will open a separate PR as it needs to be done in many backends.

Reply:
Just to be clear, this will stay unimplemented in this PR?

Reply:
Yes, it needs some stuff that will need to go in Qt also, and/or a restructuring of image directories.